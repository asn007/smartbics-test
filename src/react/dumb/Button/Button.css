:local .btn {
  border: 0;
  padding: 0;
  margin: 0;
  background: white;
  color: black;
  border: 1px solid black;
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 5px;
  padding-bottom: 5px;
  border-radius: 3px;
  transition: .2s ease all;
  cursor: pointer;
}

:local .btn:hover {
  background: black;
  color: white;
}

:local .btn::-moz-focus-inner {
  border:0;
  padding:0;
  margin:0;
}

:local .btn:disabled {
  opacity: 0.3;
  cursor: not-allowed
}

:local .btn > div {
  display: flex;
  align-items: center;
  justify-content: center;
}


:local .success {
  border-color: green;
  color: green;
}

:local .success:hover {
  background: green;
  color: white;
}

:local .danger {
  border-color: darkred;
  color: darkred;
}

:local .danger:hover {
  background: darkred;
  color: white;
}
