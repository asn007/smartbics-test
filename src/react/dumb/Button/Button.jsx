import React from 'react';
import PropTypes from 'prop-types';
import style from './Button.css';

export default function Button({
  children,
  success,
  danger,
  className,
  ...rest
}) {
  return (
    <button
      className={`${style.btn} ${success ? style.success : ''} ${danger ? style.danger : ''} ${className}`}
      {...rest}
    >
      <div>{children}</div>
    </button>
  );
}

Button.propTypes = {
  success: PropTypes.bool,
  danger: PropTypes.bool,
  className: PropTypes.string
};

Button.defaultProps = {
  success: false,
  danger: false,
  className: ''
};
