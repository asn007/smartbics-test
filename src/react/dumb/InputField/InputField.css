:local .input {
  border: 1px solid #1e1e1e;
  border-radius: 3px;
  padding: 5px;
}

:local .input:focus {
  border: 1px solid black;
}
