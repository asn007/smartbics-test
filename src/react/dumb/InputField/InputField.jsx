import React from 'react';
import PropTypes from 'prop-types';
import style from './InputField.css';

export default function InputField({ className, ...rest }) {
  return <input className={style.input + ' ' + className} {...rest} />;
}

InputField.propTypes = {
  className: PropTypes.string
};

InputField.defaultProps = {
  className: ''
};
