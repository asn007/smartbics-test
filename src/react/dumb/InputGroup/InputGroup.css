:local .inputGroup > :first-child {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  border-right-width: 0;
}

:local .inputGroup > :last-child {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-left-width: 0;
}

:local .inputGroup > :not(:last-child):not(:first-child) {
  border-radius: 0;
}
