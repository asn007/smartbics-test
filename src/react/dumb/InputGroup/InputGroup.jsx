import React from 'react';
import PropTypes from 'prop-types';
import style from './InputGroup.css';

export default function InputGroup({ children, className }) {
  return (
    <div className={style.inputGroup + ' ' + className}>
      {React.Children.map(children, child => React.cloneElement(child))}
    </div>
  );
}

InputGroup.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

InputGroup.defaultProps = {
  className: ''
};
