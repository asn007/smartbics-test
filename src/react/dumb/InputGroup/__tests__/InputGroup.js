import React from 'react';
import InputGroup from '../InputGroup';
import theme from '../InputGroup.css';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import InputField from '../../InputField';
import Button from '../../Button';

describe('InputField', () => {
  test('it should render its markup correctly', () => {
    const component = (
      <InputGroup className="test">
        <InputField value="test" />
        <Button>Test</Button>
      </InputGroup>
    );
    const shallowRendered = shallow(component);
    const reactRendered = renderer.create(component);
    expect(reactRendered.toJSON()).toMatchSnapshot();
    expect(shallowRendered.children().length).toBe(2);
    expect(shallowRendered.hasClass(theme.inputGroup)).toBe(true);
    expect(shallowRendered.hasClass('test')).toBe(true);
  });
});
