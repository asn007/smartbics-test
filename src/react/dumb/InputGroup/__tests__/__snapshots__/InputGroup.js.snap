// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField it should render its markup correctly 1`] = `
<div
  className="inputGroup test"
>
  <input
    className="input "
    value="test"
  />
  <button
    className="btn   "
  >
    <div>
      Test
    </div>
  </button>
</div>
`;
